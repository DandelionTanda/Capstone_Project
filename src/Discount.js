import { useState }  from "react";
import * as React from 'react';
import { Button, FlatList, SafeAreaView, StatusBar, StyleSheet, Text, TouchableOpacity,View,Icon,Image } from "react-native";
import 'react-native-gesture-handler';
import { NavigationContainer, getFocusedRouteNameFromRoute, } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { createStackNavigator } from '@react-navigation/stack';
import { MaterialCommunityIcons, Ionicons } from '@expo/vector-icons';
import AsyncStorage from "@react-native-async-storage/async-storage";

function getHeaderTitle(route) {
  const routeName = getFocusedRouteNameFromRoute(route) ?? 'Home';
  switch (routeName) {
    case 'Home':
      return 'Home';
    case 'Me':
      return 'Me';   
  }
}

const Tab = createBottomTabNavigator();

const Home = ( {navigation} ) => {
  const DATA = [
    {    
      name: "On-shift discount",
      value: "30%",
    },
    {   
      name: "On-shift discount",
      value: "40%",
    },
    {   
      name: "On-shift discount",
      value: "50%",
    },
    {    
      name: "On-shift discount",
      value: "60%",
    },
    {   
      name: "On-shift discount",
      value: "70%",
    },
    {   
      name: "On-shift discount",
      value: "80%",
    },
  ];
  
  const Item = ({ item, onPress, backgroundColor, textColor }) => (
    <TouchableOpacity onPress={() => navigation.navigate('Discount Details', item)} style={[styles.item, backgroundColor]}>
      <Text style={[styles.title, textColor]}>{item.name} {item.value} </Text>
    </TouchableOpacity>
  );
  const [selectedId, setSelectedId] = useState(null);

  const renderItem = ({ item, index }) => {
    const backgroundColor = index === selectedId ? '#45B8DB' : '#45B8DB';
    const color = index === selectedId ? 'white' : 'white';

    return (
      <Item
        item={item}
        onPress={() => setSelectedId(index)}
        backgroundColor={{ backgroundColor }}
        textColor={{ color }}
      />
    );
  };

  return (
    <SafeAreaView style={styles.container}>
        
      <View style={styles.clock}>
          <Text style={{fontSize: 20, color: 'white', fontWeight: 'bold'}}>
              Clocked In
          </Text>
      </View>                 
      <FlatList
        data={DATA}
        renderItem={renderItem}
        keyExtractor={(item, index) => index.toString()}        
        extraData={selectedId}
        style={{marginVertical: 25,}}
      />    
    </SafeAreaView>
    
  );
};

function Me( { navigation } ) {

  //get token
  const token =  localStorage.getItem('token')
  //get type of token
  const tokenType =  localStorage.getItem('tokenType')
  //data header
  const config = {
    method: "GET",
    headers: {Authorization: localStorage.getItem('tokenType')+ ' ' +localStorage.getItem('token')}};
  const [userName, setUserName] = useState('')
  const [photo, setPhoto] = useState('')
  function clear(){
    navigation.navigate('Login')
    AsyncStorage.clear()
    localStorage.clear()
  }
  const url = `https://my.tanda.co/api/v2/users/me`
  fetch(url,config)
  .then(res=>res.json())
   .then((res)=>{
    try{
      setUserName(res.name)
      setPhoto(res.photo)
    }
    catch(err)
    {
      console.log("err")
    }
   });
  
  return (
    <View style={{ flex: 1, justifyContent: 'center',  width:'50%', alignSelf:'center' }}>
      <Image style={{ width: '100%', height: '40%', justifyContent: 'center',alignItems: 'center', marginBottom: 30 }} 
      source={{
          uri: photo,
       }} alt = "Avatar"></Image>
      <Text style={{fontSize: 35, fontWeight: "bold", color: 'black',marginLeft:40,marginBottom:40}}>Welcome {userName}</Text>
      <TouchableOpacity
        style={styles.logoutButton}
        onPress={clear}
      >
        <Ionicons name="log-out" size={35} color="white" />     
        <Text style={{fontSize: 20, fontWeight: "bold", color: 'white',marginLeft:40}}>Log out</Text>
      </TouchableOpacity>
    </View>
  );
}


function Details( {route, navigation} ) {
    return (
      <View style={{ flex: 1, justifyContent: 'center',  alignSelf:'center'}}>
        <Text style={{fontSize: 30, fontWeight: "bold"}}>
          Name: {route.params.name}         
        </Text>
        <Text style={{fontSize: 30, fontWeight: "bold"}}> 
          Value: {route.params.value}
        </Text>
      </View>
    );
  }

function MyTabs({ navigation, route }) {
  React.useLayoutEffect(() => {
    navigation.setOptions({ headerTitle: getHeaderTitle(route) });
  }, [navigation, route]);
  return (
    <Tab.Navigator
      initialRouteName="Home"
      tabBarOptions={{
        activeTintColor: '#45B8DB',
      }}
    >
      <Tab.Screen
        name="Home"
        component={Home}
        options={{
          title: 'Home',
          tabBarLabel: 'Home',
          tabBarIcon: ({ color, size }) => (
            <MaterialCommunityIcons name="home" color={color} size={size} />
          ),
        }}
      />
      
      <Tab.Screen
        name="Me"
        component={Me}
        options={{
          tabBarLabel: 'Me',
          tabBarIcon: ({ color, size }) => (
            <MaterialCommunityIcons name="account" color={color} size={size} />
          ),
        }}
      />
      
    </Tab.Navigator>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,    
    backgroundColor: '#F5F3F3',
    
  },
  item: {
    padding: 20,
    marginVertical: 18,
    marginHorizontal: 20,
    borderTopLeftRadius: 15,
    borderTopRightRadius: 15,
    borderBottomLeftRadius: 15,
    borderBottomRightRadius: 15,
  },
  title: {
    fontSize: 30,
   
  },
  
  clock: {
    marginTop: 18,
    marginLeft: 15,
    padding: 6,
    alignSelf:  "flex-start",
    backgroundColor: "green",
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
  },
  
  logoutButton: {
    alignItems: "center",
    backgroundColor: "#E3310E",
    padding: 10,   
    flexDirection: 'row',
    borderTopLeftRadius: 15,
    borderTopRightRadius: 15,
    borderBottomLeftRadius: 15,
    borderBottomRightRadius: 15,
  },
});
const Stack = createStackNavigator();

export default function Discount() {
  return (  
      <Stack.Navigator initialRouteName="Home" >
        <Stack.Screen 
          name="Home" 
          component={MyTabs} 
          options={{ 
          tabBarLabel: 'Home!' ,
          headerStyle: {
            backgroundColor: '#45B8DB',
          },
          headerTintColor: '#fff',
          headerTitleStyle: {
            fontWeight: 'bold',
          },
          headerTitleAlign: 'center',
          headerLeft: null
        }}/>
        <Stack.Screen 
          name="Discount Details" 
          component={Details} 
          options={{           
          headerStyle: {
            backgroundColor: '#45B8DB',
          },
          headerTintColor: '#fff',
          headerTitleStyle: {
            fontWeight: 'bold',
          },
          headerTitleAlign: 'center'
        }}/>
      </Stack.Navigator>     
    
  );
}