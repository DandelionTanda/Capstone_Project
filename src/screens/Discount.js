import { useState }  from "react";
import * as React from 'react';
import { Button, FlatList, SafeAreaView, StatusBar, StyleSheet, Text, TouchableOpacity,View,Icon,Image } from "react-native";
import 'react-native-gesture-handler';
import { NavigationContainer, getFocusedRouteNameFromRoute, } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { createStackNavigator } from '@react-navigation/stack';
import { MaterialCommunityIcons, Ionicons } from '@expo/vector-icons';
import { get } from "react-native/Libraries/Utilities/PixelRatio";


export default function Discount( {route, navigation} ) {

  return ( 
    <View style={{ flex: 1, justifyContent: 'center',  alignSelf:'center'}}>
      <Text style={{fontSize: 30, fontWeight: "bold"}}>
        Name: {route.params.name}         
      </Text>
      <Text style={{fontSize: 30, fontWeight: "bold"}}> 
        Value: {route.params.value}
      </Text>
    </View>
  );
  }