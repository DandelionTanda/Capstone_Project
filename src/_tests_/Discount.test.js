import Discount from '../screens/Discount'

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer';
import {fireEvent, render, waitFor} from '@testing-library/react-native'
import {fakeUser, fakeOnDiscounts,} from './fakeData'

describe('When the user clicked on first discount item', () => {
  
  it('Should render Discount screen correctly', () => {
    const { toJSON } = render(
      <Discount route={{params: {discount:fakeOnDiscounts[0], user: fakeUser}}}/>
    );
    expect(toJSON()).toMatchSnapshot();
  })
  it('Should display clicked discont information', async () => {
    const { queryByText } = render(
      <Discount route={{params: {discount:fakeOnDiscounts[0], user: fakeUser}}}/>
    );
    await waitFor(() => {
      expect(queryByText("On-shift Discount - 1")).toBeTruthy();
      expect(queryByText("25%")).toBeTruthy();
      expect(queryByText("on-shift")).toBeTruthy();
    })
  })
  it('Should display user information', async () => {
    const { queryByText } = render(
      <Discount route={{params: {discount:fakeOnDiscounts[0], user: fakeUser}}}/>
    );
    await waitFor(() => {
      expect(queryByText("Weiran Zou")).toBeTruthy();
      expect(queryByText("zouweiran9122@gmail.com")).toBeTruthy();
      expect(queryByText("QUT Capstone - DEMO ACCOUNT")).toBeTruthy();
      expect(queryByText("1748964")).toBeTruthy();
    })
  })

  it('Should display Logo', async () => {
    const { queryByTestId } = render(
      <Discount route={{params: {discount:fakeOnDiscounts[0], user: fakeUser}}}/>
    );
    await waitFor(() => {
      expect(queryByTestId("Logo")).toBeTruthy();
      
    })
  })
})
