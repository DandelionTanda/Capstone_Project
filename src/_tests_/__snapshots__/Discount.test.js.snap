// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the user clicked on first discount item Should render Discount screen correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "flex-end",
            "height": 24,
            "marginRight": 20,
            "marginTop": 20,
            "width": 64,
          }
        }
        testID="Logo"
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#45B8DB",
          "borderRadius": 50,
          "borderWidth": 2,
          "marginBottom": 20,
          "marginTop": 30,
          "paddingBottom": 15,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 110,
          }
        }
      >
        25%
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        On-shift Discount - 1
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        on-shift
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "marginLeft": 30,
          "marginRight": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1D87E3",
            "fontSize": 22,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          Object {
            "color": "#232A22",
            "fontSize": 20,
            "lineHeight": 40,
          }
        }
      >
        Weiran Zou
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "marginLeft": 30,
          "marginRight": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1D87E3",
            "fontSize": 22,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        Email
      </Text>
      <Text
        style={
          Object {
            "color": "#232A22",
            "fontSize": 20,
            "lineHeight": 40,
          }
        }
      >
        zouweiran9122@gmail.com
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "marginLeft": 30,
          "marginRight": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1D87E3",
            "fontSize": 22,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        Employee ID
      </Text>
      <Text
        style={
          Object {
            "color": "#232A22",
            "fontSize": 20,
            "lineHeight": 40,
          }
        }
      >
        1748964
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "marginLeft": 30,
          "marginRight": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1D87E3",
            "fontSize": 22,
            "fontWeight": "bold",
            "marginTop": 10,
          }
        }
      >
        Organisation
      </Text>
      <Text
        style={
          Object {
            "color": "#232A22",
            "fontSize": 20,
            "lineHeight": 40,
          }
        }
      >
        QUT Capstone - DEMO ACCOUNT
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
